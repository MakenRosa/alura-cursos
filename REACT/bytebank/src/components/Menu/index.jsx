import React from 'react';
import styles from './Menu.module.css';

const menuOptions = [
  'Home',
  'Transfers',
  'Investiments',
  'Other services',
];

export default function Menu() {
  return (
    <nav className={styles.menu}>
      {menuOptions.map((item, index) => {
        return (
          <div key={item} className={styles.item}>
            <a href="/" className={styles.link}>
              {item}
            </a>
            {index !== menuOptions.length - 1 && (
              <div className={styles.divider} />
            )}
          </div>
        );
      })}
    </nav>
  );
}
