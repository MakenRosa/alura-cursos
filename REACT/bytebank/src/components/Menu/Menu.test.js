import { render, screen } from '@testing-library/react';
import Menu from './index';

it('should render a link to homepage', () => {
  render(<Menu />);
  const linkHome = screen.getByText('Home');
  expect(linkHome).toBeVisible();
});

it('should render a links list to homepage', () => {
  render(<Menu />);
  const linksList = screen.getAllByRole('link');
  expect(linksList).toHaveLength(4);
});

it('should not render a link to statement page', () => {
  render(<Menu />);
  const linkStatement = screen.queryByText('Statement');
  expect(linkStatement).not.toBeInTheDocument();
});

it('should render a links list with link class', () => {
  render(<Menu />);
  const linksList = screen.getAllByRole('link');
  linksList.forEach((link) => {
    expect(link).toHaveClass('link');
  });
  expect(linksList).toMatchSnapshot();
});
