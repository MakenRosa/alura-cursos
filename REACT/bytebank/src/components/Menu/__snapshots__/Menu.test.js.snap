// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a links list with link class 1`] = `
Array [
  <a
    class="link"
    href="/"
  >
    Home
  </a>,
  <a
    class="link"
    href="/"
  >
    Transfers
  </a>,
  <a
    class="link"
    href="/"
  >
    Investiments
  </a>,
  <a
    class="link"
    href="/"
  >
    Other services
  </a>,
]
`;
